bindings/rascal/representations/spherical_covariants.py

```diff
@@ -0,0 +1,223 @@
+import json
+
+from ..neighbourlist import get_neighbourlist
+from .base import RepresentationFactory, FeatureFactory
+from ..utils import get_full_name
+from ..neighbourlist.structure_manager import convert_to_structure
+from ..neighbourlist.base import NeighbourListFactory
+import numpy as np
+
+
+class SphericalCovariants(object):
+
+    """
+    Computes a SphericalCovariants representation, e.g. lambda spectrum.
+
+    Hyperparameters
+    ----------
+    interaction_cutoff : float
+        Maximum pairwise distance for atoms to be considered in
+        expansion
+
+    cutoff_smooth_width : float
+        The distance over which the the interaction is smoothed to zero
+
+    max_radial : int
+        Number of radial basis functions
+
+    max_angular : int
+        Highest angular momentum number (l) in the expansion
+
+    n_species : int
+        Number of species to be considered separately
+
+    gaussian_sigma_type : str
+        How the Gaussian atom sigmas (smearing widths) are allowed to
+        vary -- fixed ('Constant'), by species ('PerSpecies'), or by
+        distance from the central atom ('Radial').
+
+    gaussian_sigma_constant : float
+        Specifies the atomic Gaussian widths, in the case where they're
+        fixed.
+
+    soap_type : string
```

Comment:
I was planing to do thing renaming campaign after the branches are merged in master. I would also rename this on the c++ side which requires updating the reference files and many inputs.

Reply:
Right, so why not just do it now and have only one big renaming campaign (rather than this pr + another one that will need to be merged in later)? Shouldn't be that hard with reasonable search/replace tools.

Reply:
I would rather not so it now. I leave this task to you if you really want it now.

Reply:
I'm actually going back on this on the Python side; I think it would actually be better to just have one parameter called `soap_type` that can take the values `RadialSpectrum`, `PowerSpectrum`, `LambdaSpectrum`, etc., and with a Python wrapper class (`SOAP`?) that dispatches to the appropriate `Spherical[Co|In]variant` class.

I'll do the renaming on the C++ side, shouldn't take too long.

Reply:
Ok, so the _real_ reason @felixmusil didn't want to do the renaming is because the JSON scheme for these representations is going to change to a more rational scheme later (based on tensor and contraction orders). So I guess we're sticking with `soap_type` in the code until this happens.

As for the user interface, we should still keep the original names as wrappers that set the tensor and body orders appropriately; these are not intuitive parameters for anyone who's not directly working with the theory of density-based descriptors.